src/snowflake/snowpark/_internal/analyzer/materialization_utils.py

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import sys
+from enum import Enum
+
+# collections.Counter does not pass type checker. Changes with appropriate type hints were made in 3.9+
+if sys.version_info < (3, 9):
+    import collections
+    import typing
+
+    KT = typing.TypeVar("KT")
+
+    class Counter(collections.Counter, typing.Counter[KT]):
+        pass
+
+else:
+    from collections import Counter  # noqa
+
+
+class ComplexityStat(Enum):
```

Comment:
I left a comment about maybe just change this to PlanNodeCategory to be more general, and instead of having LOW_IMPACT, we can just have others,  etc

Reply:
good point